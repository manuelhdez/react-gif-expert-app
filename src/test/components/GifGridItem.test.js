import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";


describe("Pruebas en <GifGridItem />", () => {
  const title = "Mi titulo";
  const url = "http://localhost/img.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should display the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should has a paragraph with a title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should has an img tag & alt property on props', () => {
    const img = wrapper.find("img");
    //console.log(img.props());
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  })


});
