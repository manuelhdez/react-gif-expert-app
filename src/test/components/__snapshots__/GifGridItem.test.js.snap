// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should display the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Mi titulo"
    height="100"
    src="http://localhost/img.jpg"
  />
  <p>
    Mi titulo
  </p>
</div>
`;
